client/src/bookingPortal.jsx

```diff
@@ -51,26 +56,58 @@ class BookingPortal extends React.Component {
       ));
   }
 
-  handleInputDateClick() {
-    const { inputClick } = this.state;
-    if (inputClick) {
+  handleDateSelect(checkInDate, checkOutDate) {
+    this.setState({
+      checkInDate,
+      checkOutDate,
+    });
+  }
+
+  customHandleCheckInClick(checkInClick, checkOutClick) {
+    this.setState({
+      checkInClick,
+      checkOutClick,
+    });
+  }
+
+  handleCheckInClick() {
```

Comment:
If you want, you can set a variable with a switch statement to reduce the if statements and have only one this.setState.  It would look something like this:
const newState =  (()  => switch (true) {
          case (checkInClick === 'none' && checkOutClick === 'none'):
            return {checkInClick: 'block'};
           all the other cases...
        })()
this.setState({newState})